apps/studio/data/database-extensions/database-extensions-query.ts

```diff
@@ -0,0 +1,68 @@
+import { UseQueryOptions, useQuery, useQueryClient } from '@tanstack/react-query'
+import { get } from 'data/fetchers'
+import { useCallback } from 'react'
+import { ResponseError } from 'types'
+import { databaseExtensionsKeys } from './keys'
+
+export type DatabaseExtensionsVariables = {
+  projectRef?: string
+  connectionString?: string
+}
+
+export async function getDatabaseExtensions(
+  { projectRef, connectionString }: DatabaseExtensionsVariables,
+  signal?: AbortSignal
+) {
+  if (!projectRef) throw new Error('projectRef is required')
+
+  let headers = new Headers()
+  if (connectionString) headers.set('x-connection-encrypted', connectionString)
+
+  const { data, error } = await get('/platform/pg-meta/{ref}/extensions', {
+    params: {
+      header: {
+        'x-connection-encrypted': connectionString!,
+      },
+      path: {
+        ref: projectRef,
+      },
+    },
+    headers: Object.fromEntries(headers),
+    signal,
+  })
+
+  if (error) throw error
+  return data
+}
+
+export type DatabaseExtensionsData = Awaited<ReturnType<typeof getDatabaseExtensions>>
+export type DatabaseExtensionsError = ResponseError
+
+export const useDatabaseExtensionsQuery = <TData = DatabaseExtensionsData>(
+  { projectRef, connectionString }: DatabaseExtensionsVariables,
+  {
+    enabled = true,
+    ...options
+  }: UseQueryOptions<DatabaseExtensionsData, DatabaseExtensionsError, TData> = {}
+) =>
+  useQuery<DatabaseExtensionsData, DatabaseExtensionsError, TData>(
+    databaseExtensionsKeys.list(projectRef),
+    ({ signal }) => getDatabaseExtensions({ projectRef, connectionString }, signal),
+    {
+      enabled:
+        enabled && typeof projectRef !== 'undefined' && typeof connectionString !== 'undefined',
```

Comment:
Is `typeof connectionString !== 'undefined'` needed? Will break self-hosted?

Reply:
ahh thats true, self hsoted doesnt need connection string

Reply:
just to check though what about entity-types-infinite-query and database-triggers-query?

Reply:
hmm yes I think they should have it removed too! Nice catch!